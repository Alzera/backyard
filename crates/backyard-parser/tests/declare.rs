use backyard_parser::parse_eval;

#[test]
fn basic() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "declare(strict_types = 1);").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn with_body() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "declare(ticks = 1, ticks = 1) {\n}").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn with_short_body() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "declare(ticks = 1):\nenddeclare;").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}
