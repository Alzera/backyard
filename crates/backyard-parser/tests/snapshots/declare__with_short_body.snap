---
source: crates/backyard-parser/tests/declare.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: declare
    arguments:
      - node_type: declare_argument
        name:
          node_type: identifier
          name:
            - 116
            - 105
            - 99
            - 107
            - 115
          loc:
            start:
              line: 1
              column: 8
              offset: 8
            end:
              line: 1
              column: 13
              offset: 13
          leadings: ~
          trailings: ~
        value:
          node_type: number
          value:
            - 49
          loc:
            start:
              line: 1
              column: 16
              offset: 16
            end:
              line: 1
              column: 16
              offset: 16
          leadings: ~
          trailings: ~
        loc:
          start:
            line: 1
            column: 8
            offset: 8
          end:
            line: 1
            column: 16
            offset: 16
        leadings: ~
        trailings: ~
    body:
      node_type: block
      statements: []
      loc:
        start:
          line: 1
          column: 18
          offset: 18
        end:
          line: 2
          column: 0
          offset: 20
      leadings: ~
      trailings: ~
    body_type: Short
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 2
        column: 0
        offset: 20
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 2
    column: 10
    offset: 30
leadings: ~
trailings: ~
